apps/molgenis-components/src/components/tables/RoutedTableExplorer.vue

```diff
@@ -0,0 +1,223 @@
+        return parseInt(this.$route.query._page);
+      } else {
+        return 1;
+      }
+    },
+    getLimit() {
+      if (this.$route.query._limit) {
+        return parseInt(this.$route.query._limit);
+      } else {
+        return 20;
+      }
+    },
+    getView() {
+      if (this.$route.query._view) {
+        return this.$route.query._view;
+      } else {
+        return "table";
+      }
+    },
+    getConditions() {
+      let result = {};
+      Object.keys(this.$route.query).forEach((key) => {
+        if (!key.startsWith("_")) {
+          result[key] = this.$route.query[key];
+        }
+      });
+      return result;
+    },
+    updatePage(page) {
+      const query = Object.assign({}, this.$route.query);
+      query._page = page;
+      this.queryRoute(query);
+    },
+    updateOrder(order) {
+      const query = Object.assign({}, this.$route.query);
+      query._order = order.direction;
+      query._orderBy = order.column;
+      this.queryRoute(query);
+    },
+    updateView(view, limit) {
+      const query = Object.assign({}, this.$route.query);
+      query._view = view;
+      query._limit = limit;
+      delete query._page;
+      this.queryRoute(query);
+    },
+    updateLimit(limit) {
+      const query = Object.assign({}, this.$route.query);
+      query._limit = limit;
+      delete query._page;
+      this.queryRoute(query);
+    },
+    updateColumns(showColumns) {
+      const query = Object.assign({}, this.$route.query);
+      if (showColumns.length) {
+        query._col = showColumns.join(",");
+      } else {
+        delete query._col;
+      }
+      this.queryRoute(query);
+    },
+    updateAllColumns(allColumns) {
+      this.allColumns = allColumns;
+    },
+    updateFilters(showFilters) {
+      const query = Object.assign({}, this.$route.query);
+      if (showFilters.length > 0) {
+        query._filter = showFilters.join(",");
+      } else {
+        delete query._filter;
+      }
+      this.queryRoute(query);
+    },
+    updateConditions(columns) {
+      let query = Object.assign({}, this.$route.query);
+      columns.forEach((column) => {
+        const conditions = column.conditions;
+        if (conditions?.length) {
+          switch (column.columnType) {
+            case "REF":
+            case "REF_ARRAY":
+            case "REFBACK":
+            case "ONTOLOGY":
+            case "ONTOLOGY_ARRAY":
+              query[column.name] = JSON.stringify(conditions);
+              break;
+            case "DATE":
+            case "DATETIME":
+            case "INT":
+            case "DECIMAL":
+              query[column.name] = conditions
+                .map((v) => v.join(".."))
+                .join(",");
+              break;
+            default:
+              query[column.name] = conditions.join(",");
+          }
+        } else {
+          delete query[column.name];
+        }
+      });
+      this.queryRoute(query);
+    },
+    queryRoute(query) {
```

Comment:
queryRoute ? would suggest something like updateRoute  or updateRouteQuery or updateUrl , this a method ( action ) the has a explicit side effect